tests/integration/mutation/create/with_default_values_test.go

```diff
@@ -0,0 +1,259 @@
+						name: String @default(string: "Bob")
+						age: Int @default(int: 40)
+						points: Float @default(float: 10)
+						metadata: JSON @default(json: "{\"one\":1}")
+						image: Blob @default(blob: "ff0099")
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				DocMap: map[string]any{
+					"age":      nil,
+					"active":   nil,
+					"name":     nil,
+					"points":   nil,
+					"created":  nil,
+					"metadata": nil,
+					"image":    nil,
+				},
+			},
+			testUtils.Request{
+				Request: `query {
+					Users {
+						age
+						active
+						name
+						points
+						created
+						metadata
+						image
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"age":      nil,
+							"active":   nil,
+							"name":     nil,
+							"points":   nil,
+							"created":  nil,
+							"metadata": nil,
+							"image":    nil,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestMutationCreate_WithDefaultValues_ValuesProvided_SetsValue(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple create mutation, with default values and values provided",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						active: Boolean @default(bool: true)
+						created: DateTime @default(dateTime: "2000-07-23T03:00:00-00:00")
+						name: String @default(string: "Bob")
+						age: Int @default(int: 40)
+						points: Float @default(float: 10)
+						metadata: JSON @default(json: "{\"one\":1}")
+						image: Blob @default(blob: "ff0099")
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				DocMap: map[string]any{
+					"age":      int64(50),
+					"active":   false,
+					"name":     "Alice",
+					"points":   float64(5),
+					"created":  time.Time(time.Date(2024, time.June, 18, 1, 0, 0, 0, time.UTC)),
+					"metadata": "{\"two\":2}",
+					"image":    "aabb33",
+				},
+			},
+			testUtils.Request{
+				Request: `query {
+					Users {
+						age
+						active
+						name
+						points
+						created
+						metadata
+						image
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"age":      int64(50),
+							"active":   false,
+							"name":     "Alice",
+							"points":   float64(5),
+							"created":  time.Time(time.Date(2024, time.June, 18, 1, 0, 0, 0, time.UTC)),
+							"metadata": "{\"two\":2}",
+							"image":    "aabb33",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestMutationCreate_WithDefaultValue_NoValueProvided_CreatedTwice_ReturnsError(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple create mutation, with default value, no value provided, and created twice",
+		SupportedMutationTypes: immutable.Some([]testUtils.MutationType{
+			// This test will fail if using the collection save
+			// method because it does not create two unique docs
```

Comment:
question: I can't spot anything super obvious in the test code as to why `Save` would not work, would you mind expanding a little?

Reply:
It seems like calling save does not actually create two unique documents when the fields have the same value. I'm guessing it checks if the docID exists and calls update instead of create.

Reply:
Oh true - it does - thanks Keenan.  Might be worth adding that to this code-comment then.